spec/models/miq_expression_spec.rb

```diff
@@ -350,6 +350,112 @@
       expect { exp.to_ruby }.to raise_error(RuntimeError, "Ruby scripts in expressions are no longer supported. Please use the regular expression feature of conditions instead.")
     end
 
+    it "generates the ruby for a STARTS WITH expression" do
+      actual = described_class.new("STARTS WITH" => {"field" => "Vm-name", "value" => "foo"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> =~ /^foo/"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an ENDS WITH expression" do
+      actual = described_class.new("ENDS WITH" => {"field" => "Vm-name", "value" => "foo"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> =~ /foo$/"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an AND expression" do
+      actual = described_class.new("AND" => [{"=" => {"field" => "Vm-name", "value" => "foo"}},
+                                             {"=" => {"field" => "Vm-vendor", "value" => "bar"}}]).to_ruby
+      expected = "(<value ref=vm, type=string>/virtual/name</value> == \"foo\" and <value ref=vm, type=string>/virtual/vendor</value> == \"bar\")"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an OR expression" do
+      actual = described_class.new("OR" => [{"=" => {"field" => "Vm-name", "value" => "foo"}},
+                                            {"=" => {"field" => "Vm-vendor", "value" => "bar"}}]).to_ruby
+      expected = "(<value ref=vm, type=string>/virtual/name</value> == \"foo\" or <value ref=vm, type=string>/virtual/vendor</value> == \"bar\")"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for a NOT expression" do
+      actual = described_class.new("NOT" => {"=" => {"field" => "Vm-name", "value" => "foo"}}).to_ruby
+      expected = "!(<value ref=vm, type=string>/virtual/name</value> == \"foo\")"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for a ! expression" do
+      actual = described_class.new("!" => {"=" => {"field" => "Vm-name", "value" => "foo"}}).to_ruby
+      expected = "!(<value ref=vm, type=string>/virtual/name</value> == \"foo\")"
+      expect(actual).to eq(expected)
+    end
+
+    it "generates the ruby for an IS NULL expression" do
+      actual = described_class.new("IS NULL" => {"field" => "Vm-name"}).to_ruby
+      expected = "<value ref=vm, type=string>/virtual/name</value> == \"\""
```

Comment:
Yeah, that's correct, even though it's weird. The quote method only cares about the data type. So if it's a string and the value is nil it returns and empty string. Would be good to do the same with an integer, float or date field.